### PR TITLE
FacadeAct tests

### LINKED ISSUE #441: Complete `FacadeAct` coverage tests
* The new version of `FacadeAct` has an uncovered line and a few branches that need a formal test.

### PR DESCRIPTION
**Pull request by julianmrodri**
* Adds new tests for branch coverage in `FacadeAct.sol`

* Adds checks in `FacadeAct.sol` to only do trading from BackingManager if the basket is not disabled. This avoids unexpected reverts when `manageTokens` is called.

             
Closes #441

### CODE REVIEW

File: contracts/facade/FacadeAct.sol [hunk 1]
```diff
@@ -79,116 +79,128 @@ contract FacadeAct is IFacadeAct {
...
-                        return (
-                            address(cache.rTokenTrader),
-                            abi.encodeWithSelector(
-                                cache.rTokenTrader.settleTrade.selector,
-                                erc20s[i]
-                            )
-                        );
-                    }
+            } else if (cache.bh.status() != CollateralStatus.DISABLED) {
+                // Only trade if basket is sound
```
Comment (tbrent) [inline contracts/facade/FacadeAct.sol:83]: Nit: comment should be "Only trade if basket is non-DISABLED"

Comment (tbrent) [reply contracts/facade/FacadeAct.sol:83]: We expect there to be times where 2 tokens default slightly offset, such that the first `refreshBasket` results in IFFY

Comment (julianmrodri) [reply contracts/facade/FacadeAct.sol:83]: Totally, will fix it

File: contracts/facade/FacadeAct.sol [hunk 2]
```diff
@@ -79,116 +79,128 @@ contract FacadeAct is IFacadeAct {
...
+                    for (uint256 i = 0; i < erc20s.length; i++) {
+                        // rTokenTrader: if there's a trade to settle
+                        ITrade trade = cache.rTokenTrader.trades(erc20s[i]);
+                        if (address(trade) != address(0) && trade.canSettle()) {
+                            // cache.rTokenTrader.settleTrade(...)
                             return (
                                 address(cache.rTokenTrader),
                                 abi.encodeWithSelector(
-                                    cache.rTokenTrader.manageToken.selector,
+                                    cache.rTokenTrader.settleTrade.selector,
```
[the remark this answers is not in the document]
Comment (julianmrodri) [reply contracts/facade/FacadeAct.sol]: Refactored this, only added the check for those sections that require `manageTokens` (which are the recollateralization and the sweeping revenue from Backing Manager) The rest remains the same.

File: contracts/facade/FacadeAct.sol [hunk 3]
```diff
@@ -79,7 +79,9 @@ contract FacadeAct is IFacadeAct {
                         );
                     }
                 }
-            } else if (!cache.bh.fullyCollateralized()) {
+            } else if (
```
Comment (tbrent) [inline contracts/facade/FacadeAct.sol:82]: Actually this condition should be status == CollateralStatus.SOUND. Sorry for not catching that on the first pass through. BackingManager.manageTokens reverts when IFFY

Comment (julianmrodri) [reply contracts/facade/FacadeAct.sol:82]: So the check should be `if (cache.bh.status() == CollateralStatus.SOUND && !cache.bh.fullyCollateralized()) {` Is it the same for the call below? in Line 155?

**Pull request merged**